liquigraph-core/src/main/java/org/liquigraph/core/configuration/ConfigurationBuilder.java

```diff
@@ -165,7 +164,19 @@ public ConfigurationBuilder withRunMode() {
      * @return itself for chaining purposes
      */
     public ConfigurationBuilder withDryRunMode(Path outputDirectory) {
-        this.executionMode = new DryRunMode(outputDirectory);
+        return withExecutionMode(new DryRunMode(outputDirectory));
+    }
+
+    /**
+     * Manually sets execution mode of liquigraph.
+     *
+     * This cas be RunMode, DryRunMode or any other kind of executionMode
+     *
+     * @param executionMode the execution mode to run
+     * @return itself for chaining purposes
+     */
+    public ConfigurationBuilder withExecutionMode(ExecutionMode executionMode) {
```

Comment:
can probably become private (as there are no new execution modes anymore)

Reply:
done in commit eee1086

I reverted this change